### PR TITLE
Refactor: Update recurring support for PayPal Donations

### PR DESCRIPTION
**Pull request by kjohnson**
## Description

This PR updates the recurring implementation of the PayPal Donations gateway (now that recurring support is available in the `develop` branch of next gen).

- PayPal Provider Script is now set to defer loading.
- Extracted a `<PaymentMethodsWrapper/>` component for observing form changes.
- Visually hide Hosted Fields when not supported (prevent re-mounting error).
- Smart Buttons now conditionally set `createOrder()` or `createSubscription()`.

## Visuals

[image](https://www.loom.com/share/cbe6915cd9c04a459367f5a890052d55)

## Testing Instructions

Toggle between one-time/recurring donations.
The PayPal Donations gateway should not throw an error.

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@kjohnson when I click the PayPal button for a recurring donation the request fails.  Can you please confirm this gateway is functional when submitting a recurring donation.

**Comment by kjohnson** [comment]
image

File: src/NextGen/Gateways/PayPalCommerce/PayPalCommerceSubscriptionModule.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        $subscriptionId = $gatewayData['payPalSubscriptionId'];
+
+        return new SubscriptionProcessing($subscriptionId);
+    }
+
+    public function cancelSubscription(Subscription $subscription)
+    {
+        // TODO: Implement cancelSubscription() method.
+    }
```
Comment (jonwaldstein) [inline src/NextGen/Gateways/PayPalCommerce/PayPalCommerceSubscriptionModule.php:31]: @kjohnson You will need to add this logic for cancelling a subscription like in the [PayPal standard module](https://github.com/impress-org/givewp-next-gen/blob/6ccb343f08ab5e0d8fa52dbdfb5ca56ef5e76138/src/NextGen/Gateways/PayPal/PayPalStandardGateway/PayPalStandardGatewaySubscriptionModule.php#L205).

You can test this out in the donor dashboard by clicking cancel subscription on an active subscription and make sure it actually gets cancelled in PayPal.

This should be the last piece to this integration as everything else appears to be working properly.

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: src/NextGen/Gateways/PayPalCommerce/PayPalCommerceSubscriptionModule.php (hunk 1 above)
Comment (kjohnson) [reply src/NextGen/Gateways/PayPalCommerce/PayPalCommerceSubscriptionModule.php:31]: Ah, the donor dashboard. I had tested via the Subscription details screen, which already works with the gateway.

Comment (kjohnson) [reply src/NextGen/Gateways/PayPalCommerce/PayPalCommerceSubscriptionModule.php]: ## Call Summary

The PayPal Commerce recurring gateway was hooked in to the GiveWP Subscription status change and synced that new status with PayPal.

Due to changes in the Recurring add-on, this cancellation needed to be defined in the Subscription Module.

The solution was to call the PayPal Commerce Subscription Repository `updateStatus()` method from the subscription module cancel method.

**Review by jonwaldstein — APPROVED** [review]

**Pull request merged**